src/NHapi.Base/Parser/PipeParser.cs

```diff
@@ -445,15 +452,19 @@ public virtual void Parse(ISegment destination, string segment, EncodingCharacte
             }
 
             // set data type of OBX-5
-            if (destination.GetType().FullName.IndexOf("OBX") >= 0)
+            if (destination.GetType().FullName?.IndexOf("OBX", StringComparison.Ordinal) >= 0)
```

Comment:
this would have thrown an exception before, if `FullName` was null... does this matter?

Reply:
You are correct, will undo this change.

Reply:
> Gets the fully qualified name of the type, including its namespace but not its assembly.
Returns:
The fully qualified name of the type, including its namespace but not its assembly; or null if the current instance represents a generic type parameter, an array type, pointer type, or byref type based on a type parameter, or a generic type that is not a generic type definition but contains unresolved type parameters.

In this case it should be fine since `destination` is always of type `ISegment`, is not a generic instance, is not passed byref and is not an array type etc, so should never be null.

Will change to `!.`